### PR TITLE
[ticket/13689] Allow modifying the query for postgres native fulltext search

### PR DESCRIPTION
**Pull request by brunoais**
- [ ] based on https://github.com/phpbb/phpbb/pull/3479

https://tracker.phpbb.com/browse/PHPBB3-13689

### CODE REVIEW

File: phpBB/phpbb/search/fulltext_postgres.php [hunk 1]
```diff
@@ -409,6 +409,52 @@ public function keyword_search($type, $fields, $terms, $sort_by_sql, $sort_key,
...
+			'sql_sort_join',
+			'sort_days',
+			'sql_match',
+			'sql_match_where',
+			'sort_by_sql',
+			'sort_key',
+			'sort_dir',
+			'sql_sort',
+			'start',
+			);
```
Comment (nickvergessen) [inline phpBB/phpbb/search/fulltext_postgres.php]: -1 tab

File: phpBB/phpbb/search/fulltext_postgres.php [hunk 2]
```diff
@@ -409,6 +417,55 @@ public function keyword_search($type, $fields, $terms, $sort_by_sql, $sort_key,
...
+		*									Used in conjunction with sql_sort_table
+		* @var	int		sort_days			Time, in days, of the oldest possible post to list
+		* @var	string	sql_match			Which columns to do the search on.
+		* @var	string	sql_match_where		Extra conditions to use to properly filter the matching process
+		* @var	string	sort_by_sql			The possible predefined sort types
+		* @var	string	sort_key			The sort type used from the possible sort types
+		* @var	string	sort_dir			"a" for ASC or "d" dor DESC for the sort order used
+		* @var	string	sql_sort			The result SQL when processing sort_by_sql + sort_key + sort_dir
+		* @var	int		start				How many posts to skip in the search results (used for pagination)
+		* @since 3.1.4-RC1
```
Comment (marc1706) [inline phpBB/phpbb/search/fulltext_postgres.php]: 3.1.5-RC1

**Comment by nickvergessen** [comment]
Based on https://github.com/phpbb/phpbb/pull/3479

File: phpBB/develop/search_fill.php [hunk 3]
```diff
@@ -40,7 +40,7 @@
 }
 
 $error = false;
-$search = new $search_type($error, $phpbb_root_path, $phpEx, $auth, $config, $db, $user);
+$search = new $search_type($error, $phpbb_root_path, $phpEx, $auth, $config, $db, $phpbb_dispatcher, $user);
```
Comment (Nicofuma) [inline phpBB/develop/search_fill.php]: at the end please

File: phpBB/includes/acp/acp_main.php [hunk 4]
```diff
@@ -632,7 +632,7 @@ function main($id, $mode)
 		{
 			$error = false;
 			$search_type = $config['search_type'];
-			$search = new $search_type($error, $phpbb_root_path, $phpEx, $auth, $config, $db, $user);
+			$search = new $search_type($error, $phpbb_root_path, $phpEx, $auth, $config, $db, $phpbb_dispatcher, $user);
```
Comment (Nicofuma) [inline phpBB/includes/acp/acp_main.php]: at the end please

File: phpBB/includes/acp/acp_search.php [hunk 5]
```diff
@@ -608,7 +608,7 @@ function init_search($type, &$search, &$error)
 		}
 
 		$error = false;
-		$search = new $type($error, $phpbb_root_path, $phpEx, $auth, $config, $db, $user);
+		$search = new $type($error, $phpbb_root_path, $phpEx, $auth, $config, $db, $phpbb_dispatcher, $user);
```
Comment (Nicofuma) [inline phpBB/includes/acp/acp_search.php]: at the end please

File: phpBB/includes/functions_admin.php [hunk 6]
```diff
@@ -920,7 +920,7 @@ function delete_posts($where_type, $where_ids, $auto_sync = true, $posted_sync =
 	}
 
 	$error = false;
-	$search = new $search_type($error, $phpbb_root_path, $phpEx, $auth, $config, $db, $user);
+	$search = new $search_type($error, $phpbb_root_path, $phpEx, $auth, $config, $db, $phpbb_dispatcher, $user);
```
Comment (Nicofuma) [inline phpBB/includes/functions_admin.php]: at the end please

File: phpBB/includes/functions_posting.php [hunk 7]
```diff
@@ -2209,7 +2209,7 @@ function submit_post($mode, $subject, $username, $topic_type, &$poll, &$data, $u
 		}
 
 		$error = false;
-		$search = new $search_type($error, $phpbb_root_path, $phpEx, $auth, $config, $db, $user);
+		$search = new $search_type($error, $phpbb_root_path, $phpEx, $auth, $config, $db, $phpbb_dispatcher, $user);
```
Comment (Nicofuma) [inline phpBB/includes/functions_posting.php]: at the end please

File: phpBB/includes/mcp/mcp_main.php [hunk 8]
```diff
@@ -1197,7 +1197,7 @@ function mcp_fork_topic($topic_ids)
 				}
 
 				$error = false;
-				$search = new $search_type($error, $phpbb_root_path, $phpEx, $auth, $config, $db, $user);
+				$search = new $search_type($error, $phpbb_root_path, $phpEx, $auth, $config, $db, $phpbb_dispatcher, $user);
```
Comment (Nicofuma) [inline phpBB/includes/mcp/mcp_main.php]: at the end please

File: phpBB/includes/mcp/mcp_post.php [hunk 9]
```diff
@@ -497,7 +497,7 @@ function change_poster(&$post_info, $userdata)
 	{
 		// We do some additional checks in the module to ensure it can actually be utilised
 		$error = false;
-		$search = new $search_type($error, $phpbb_root_path, $phpEx, $auth, $config, $db, $user);
+		$search = new $search_type($error, $phpbb_root_path, $phpEx, $auth, $config, $db, $phpbb_dispatcher, $user);
```
Comment (Nicofuma) [inline phpBB/includes/mcp/mcp_post.php]: at the end please

**Pull request merged**